output/elastic/outputelastic_test.go

```diff
@@ -20,6 +24,35 @@ func init() {
 	config.RegistOutputHandler(ModuleName, InitHandler)
 }
 
+func Test_ResolveVars(t *testing.T) {
+	a := assert.New(t)
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		fmt.Printf("%v\n", r)
+		w.WriteHeader(200)
+	}
+	ts := httptest.NewServer(http.HandlerFunc(handler))
+	defer ts.Close()
+
+	err := os.Setenv("MYVAR", ts.URL)
+	a.Nil(err)
+	conf, err := config.LoadFromYAML([]byte(strings.TrimSpace(`
+debugch: true
+output:
+  - type: elastic
+    url: ["%{MYVAR}"]
+    index: "gogstash-index-test"
+    document_type: "testtype"
+    document_id: "%{fieldstring}"
+    bulk_actions: 0
+	`)))
+	a.Nil(err)
+	a.NotNil(conf)
+	_, err = InitHandler(context.TODO(), &conf.OutputRaw[0])
```

Comment:
Could you check the config will resolve to `ts.URL` successfully after here?

Reply:
I'm not sure.
Do you mean something like this?

```suggestion
c, err := InitHandler(context.TODO(), &conf.OutputRaw[0])
a.Nil(err)
outputConfig := c.(*OutputConfig)
a.Equal(ts.URL, outputConfig.URL)
```
the problem is I'm not changing the original url config string as you suggested, so not sure how to actually check it. Elastic client doesn't have anything that I can see allowing access of the url either.

Reply:
Oh, sorry, It will not save resolved url to config now, I made a mistake.

Reply:
But maybe we could create a function for resolving urls and test that function? Do you think it is necessary?

Reply:
if FormatWithEnv is already tested I don't see the need of testing this again (which is already partially tested by resolving MYVAR on the test)
If you have any suggestions I'm happy to change whatever needed.